File: mog-rs/factory/snowflake-tables-to-python-dataclass/tests/expected.py
from __future__ import annotations
from dataclasses import dataclass
from datetime import date, datetime, time
from decimal import Decimal
from typing import Any

@dataclass
class Subscriber:
    subscriber_id: int
    first_name: str
    last_name: str
    updated_at: datetime

@dataclass
class CampaignRun:
    campaign_run_id: int
    account_code: str
    email: str | None
    is_active: bool
    signed_up_on: date
    notes: str | None
    balance: Decimal # TODO(mog): Snowflake fixed-point column; enable the connector's exact-value option so it does not approximate.
    last_seen_at: datetime | None

@dataclass
class PageView:
    event_id: int
    seq_no: int
    retry_count: int | None
    amount: Decimal # TODO(mog): Snowflake fixed-point column; enable the connector's exact-value option so it does not approximate.
    rate: float | None
    is_sampled: bool
    event_date: date
    ingested_at: datetime
    occurred_at: datetime
    synced_at: datetime | None
    source_name: str
    raw_body: str | None
    payload: dict[str, Any] | None # TODO(mog): Snowflake semi-structured column; the connector hands back raw JSON unless you parse it.
    tags: list[Any] | None # TODO(mog): Snowflake list column has no declared element type; narrow it yourself.
    context: dict[str, Any] | None # TODO(mog): Snowflake semi-structured column; the connector hands back raw JSON unless you parse it.
    checksum: bytes | None

File: mog-rs/factory/snowflake-tables-to-typescript/tests/expected.ts
export interface WebSession {
  webSessionId: number; // TODO(mog): Snowflake column can exceed JavaScript's safe range of 2^53; carry wide values losslessly.
  firstName: string;
  lastName: string;
  updatedAt: string;
}

export interface CustomerProfile {
  customerProfileId: number; // TODO(mog): Snowflake column can exceed JavaScript's safe range of 2^53; carry wide values losslessly.
  accountCode: string;
  email: string | null;
  isActive: boolean;
  signedUpOn: string;
  notes: string | null;
  balance: number; // TODO(mog): Snowflake fixed-point column; JavaScript cannot hold every scale exactly -- use an exact-arithmetic library.
  lastSeenAt: string | null;
}

export interface TelemetryEvent {
  eventId: number; // TODO(mog): Snowflake column can exceed JavaScript's safe range of 2^53; carry wide values losslessly.
  seqNo: number;
  retryCount: number | null;
  amount: number; // TODO(mog): Snowflake fixed-point column; JavaScript cannot hold every scale exactly -- use an exact-arithmetic library.
  rate: number | null;
  isSampled: boolean;
  eventDate: string;
  ingestedAt: string;
  occurredAt: string;
  syncedAt: string | null;
  sourceName: string;
  rawBody: string | null;
  payload: unknown | null; // TODO(mog): Snowflake semi-structured column; narrow the loose mapping with a type guard or a schema validator.
  tags: unknown[] | null; // TODO(mog): Snowflake list column has no declared element type; set the element type yourself.
  context: unknown | null; // TODO(mog): Snowflake semi-structured column; narrow the loose mapping with a type guard or a schema validator.
  checksum: Uint8Array | null;
}
